### PR TITLE
Caching + geocoding + new themes

### PR DESCRIPTION
**Pull request by pokyah**
Hello @koenderks ,

I have implemented the following new functionalities : 

1. a caching mechanism using `memoise `package to alleviate calls to OSM servers and allowing to rebuild the same map with a new theme much faster. It does not change much the whole package. The magic happens thanks to a wrapper inside the exposed `cityview` function which has now 3 new params : `cache`, `persistent_cache `and `clear_cache`

2. a geocoding feature based on `tidygeocoder` package. If the city name is not found in the `rcityview` database, the `cityview` function will now attempt to geocode the name thanks to `tidygeocoder`. Implemented a new param  `method` in  `cityview`
 function to let the user choose the geocoding method provided to  `tidygeocoder`. Defaults to  `osm`.
 
3. have implemented new themes inspired from pyton prettymaps : https://github.com/marceloprates/prettymaps/tree/main. Added in the `helpers.R` script.

Let me know if something is unclear or need some adjustments. I'll be happy to contribute.

Best regards

### CODE REVIEW

File: .Rbuildignore
Comment (koenderks) [inline .Rbuildignore:1]: The changes to this file can be removed

File: R/builders.R
Comment (koenderks) [inline R/builders.R:1]: The additional spaces in this file can be removed

File: R/cityview.R [hunk 1]
```diff
@@ -1,32 +1,23 @@
-# Copyright (C) 2022-2022 Koen Derks
+library(dplyr)
```
Comment (koenderks) [inline R/cityview.R:1]: Please remove these library calls, see section 'Dependencies' at https://cran.r-project.org/web/packages/EpiForsk/vignettes/Writing_functions.html. Use `::` to call functions from a difference package.

File: R/cityview.R [hunk 2]
```diff
@@ -200,79 +95,77 @@ cityview <- function(name = NULL,
...
+  themeOptions <- .themeOptions(theme)
   border <- match.arg(border)
-  ticks <- 61 + as.numeric(!is.null(halftone)) + as.numeric(places > 0)
-  # Look up city ###############################################################
-  city <- .getCity(name)
-  if (is.null(city)) {
-    return(invisible())
+  
+  # Persistent cache directory setup
+  cache_dir <- file.path(path.expand("~"), "Documents", "cityview_cache")
```
Comment (koenderks) [inline R/cityview.R:110]: Did you look into using a `tempdir()`? I'm not sure if that would work btw.

File: R/cityview.R [hunk 3]
```diff
@@ -200,79 +95,77 @@ cityview <- function(name = NULL,
...
-  )
-  # Error handling #############################################################
-  if ("try-error" %in% class(try)) {
-    if (try[[1]] == "Error in resp_abort(resp, error_body(req, resp)) : \n  HTTP 504 Gateway Timeout.\n") {
-      stop("The overpass server is not able to respond to your request, traffic might be too high.")
+  
+  # Create cached buildCity function
+  buildCity_func <- if (cache) {
+    if (persistent_cache) {
+      memoise(.buildCity, cache = cache_filesystem(cache_dir))
```
Comment (koenderks) [inline R/cityview.R:128]: See comment above, this should become`memoise::memoise`

File: R/helpers.R [hunk 4]
```diff
@@ -13,6 +13,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+
```
Comment (koenderks) [inline R/helpers.R:16]: Extra line?

File: rcityviews.Rproj
Comment (koenderks) [inline rcityviews.Rproj:1]: This file can be removed from the PR

File: inst/fonts/LibreBaskerville-Regular.ttf
Comment (koenderks) [inline inst/fonts/LibreBaskerville-Regular.ttf:1]: Should you add these fonts in the `onLoad()` function in the `zzz.R` file?

File: DESCRIPTION
Comment (koenderks) [inline DESCRIPTION:1]: Do we use `dplyr`, `httr` and `jsonlite` explicitly in the package (i.e., do they need to be imported, or are they dependencies of other packages)?

**Review by koenderks — CHANGES_REQUESTED** [review]
Hi, 

From my first very short initial check, I noticed the following global stuff (see comments). If you could adjust that, then I can hopefully check out the code on a more detailed level somewhere next week. 

Additionally, it seems that the following code does not use the caching feature, or maybe I misunderstand the intended workflow?
```
p <- rcityviews::cityview(name = "Hue", theme = "vintage")
p <- rcityviews::cityview(name = "Hue", theme = "original")
```
I think for this caching style to work, you would need to separate the data requesting from the `.buildCity()` function, which means quite some rewriting.

Finally, could you also run `roxygen2::roxygenize()` and `styler::style_package()` on the package?

**Comment by koenderks** [comment]
I have implemented the caching implementation on https://github.com/koenderks/rcityviews/pull/28, such that the following command does not require re-requesting the data from the overpass server. 
```
p <- cityview("Hue", theme = "original")
p <- cityview("Hue", theme = "rouge")
```
We can add your further changes like the geocoding, but it may be a better idea to rebase and make a new PR with that functionality. I would prefer to separate the individual components for version control, so its probably best to make a PR with only the geocoding functionality first.

**Comment by pokyah** [comment]
> I have implemented the caching implementation on #28, such that the following command does not require re-requesting the data from the overpass server.
> 
> ```
> p <- cityview("Hue", theme = "original")
> p <- cityview("Hue", theme = "rouge")
> ```
> 
> We can add your further changes like the geocoding, but it may be a better idea to rebase and make a new PR with that functionality. I would prefer to separate the individual components for version control, so its probably best to make a PR with only the geocoding functionality first.

Thanks for this ! I'll do the rebase and submit a new PR for the geocoding stuff

**Pull request closed without merging**